libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -855,21 +862,29 @@ void AP_GPS::update_instance(uint8_t instance)
         data_should_be_logged = true;
     }
 
-#if GPS_MAX_RECEIVERS > 1
+#if GPS_MOVING_BASELINE
     if (drivers[instance] && _type[instance] == GPS_TYPE_UBLOX_RTK_BASE) {
         // see if a moving baseline base has some RTCMv3 data
         // which we need to pass along to the rover
         const uint8_t *rtcm_data;
         uint16_t rtcm_len;
+        bool clear_rtcmv3_buffer = false;
         if (drivers[instance]->get_RTCMV3(rtcm_data, rtcm_len)) {
+            if (_mbl_data_cb != nullptr) {
```

Comment:
i think this is simpler by exposing clear_RTCMv3() on AP_GPS, and then AP_Periph calls gps.get_RTCMv3(instance, rtcm_data, rtcm_len) and then calls the clear